app/src/androidTest/java/ch/epfl/culturequest/ArtDescriptionDisplayActivityTest1.java

```diff
@@ -124,6 +133,35 @@ public void activityDisplayingPostButton() {
         onView(withId(R.id.post_button)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
     }
 
+    @Test
+    public void activityDisplayingShareButton() {
+        onView(withId(R.id.share_button)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+    }
+
+    @Test
+    public void checkSharingSendsProperIntent() {
+        Intents.init();
+        onView(withId(R.id.artName)).perform(swipeUp());
+        onView(withId(R.id.artSummary)).perform(swipeUp(), swipeUp(), swipeUp(), swipeUp());// Scroll to the bottom of the RecyclerView
+        onView(withId(R.id.share_button)).perform(click());
+        intended(hasAction(Intent.ACTION_CHOOSER));
+
+        Matcher<Intent> expectedIntent = Matchers.allOf(
+                hasAction(Intent.ACTION_SEND),
+                hasExtra(Intent.EXTRA_TEXT, "I just scanned Mona Lisa with \uD835\uDC02\uD835\uDC2E\uD835\uDC25\uD835\uDC2D\uD835\uDC2E\uD835\uDC2B\uD835\uDC1E\uD835\uDC10\uD835\uDC2E\uD835\uDC1E\uD835\uDC2C\uD835\uDC2D!\n\nIt's a epic artwork from Da Vinci, displayed at Louvre, Paris.\n\nDownload the app here: https://play.google.com/store/apps/details?id=com.culturequest.culturequest"),
```

Comment:
Is Cultute Quest encoded in hexadecimal so that you can display it in bold ?

Reply:
Yes I display it in stylized characters (was not able to do the same with ** chars as some messaging apps do not stylize chars when a photo is sent)